### ISSUE TITLE
favorites enhancement

### ISSUE THREAD

**Issue by h00die** [rapid7/metasploit-framework#17410]
@ErikWynter since you wrote the original module #14201.

I always forget `exploit/multi/script/web_delivery`'s name, so I thought adding it to my favorite list would make a lot of sense. So i typed `favorite` and it tells you how to add and delete items:

```
msf6 > favorite
[-] No module has been provided to favorite.
Usage: favorite [mod1 mod2 ...]

Add one or multiple modules to the list of favorite modules stored in /root/.msf4/fav_modules
If no module name is specified, the command will add the active module if there is one

OPTIONS:

    -c  Clear the contents of the favorite modules file
    -d  Delete module(s) or the current active module from the favorite modules file
    -h  Help banner
```

Great, so I add that module:

```
msf6 > favorite exploit/multi/script/web_delivery
[+] Added exploit/multi/script/web_delivery to the favorite modules file
```

However, at this point I hit a 'so now what' moment.  It was added, and I know how to delete it, but not how to show/list/use it.

I suggest adding to the module description an instruction such as 'show favorites to list your favorites!'

**Comment by h00die** [comment]
Or maybe change it to:

`favorite -a [mod1, mod2]`

and the default behavior of `favorite` to list them?  

Or add a `favorite -l` or `-s` to show?   Just some thoughts

**Comment by jmartin-r7** [comment]
In the original PR details `show favorites` is listed as the way to find the list, I do agree this may not be as intuitive as I originally thought it to be since `options` and `advanced` don't require the `show` command to list the current state.

I like the idea to hint in `help` when an item is added, another idea is to have the plural form show the list `favorites`.

**Comment by ErikWynter** [comment]
@h00die @jmartin-r7 thanks for the feedback! I can definitely understand the confusion and I agree it would be good to update the help menu. When it comes to improving the way the list of favorites can be obtained, which option would you prefer:
1. add `favorites` in addition to `show favorites` and update the `favorite -h` info to explicitly mention this
2. remove support for `show favorites` and instead add `favorite -l` to do the same
3. add `favorites` in addition to `show favorites` AND add `favorite -l` to do the same

I would personally go for 1 or 3 since I want to avoid removing a command that some folks may be used to by now.

**Comment by h00die** [comment]
1 or 3 sounds good to me

**Comment by gwillcox-r7** [comment]
Closed by #17504
